lib/active_storage/service/mirror_service.rb

```diff
@@ -17,9 +18,7 @@ def initialize(primary:, mirrors:)
   end
 
   def upload(key, io, checksum: nil)
-    each_service.collect do |service|
-      service.upload key, io.tap(&:rewind), checksum: checksum
-    end
+    perform_across_services :upload, key, io.tap(&:rewind), checksum: checksum
```

Comment:
This isn’t going to work. We can’t share a single IO between concurrent uploads.